ddp-workspace/projects/ddp-dsm-ui/src/app/guards/study.guard.ts

```diff
@@ -0,0 +1,28 @@
+import {Injectable} from '@angular/core';
+import {Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot} from '@angular/router';
+import {Auth} from '../services/auth.service';
+import {ComponentService} from '../services/component.service';
+import {SessionService} from '../services/session.service';
+
+@Injectable({providedIn: 'root'})
+export class StudyGuard implements CanActivate {
+
+  constructor(private router: Router) {
+  }
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean {
+    const selectedRealm = localStorage.getItem(ComponentService.MENU_SELECTED_REALM);
+
+    if(this.allowAccessToStudy(selectedRealm, state)) return true;
+
+    this.router.navigate([selectedRealm ? selectedRealm : '']);
```

Comment:
**#minor #optional**

`selectedRealm ? selectedRealm : ''` can be shortened like `selectedRealm || ''`

Reply:
Right! Thanks!